src/eth/TransactionManager.js

```diff
@@ -12,12 +15,69 @@ export default class TransactionManager extends PublicService {
   }
 
   formatHybridTx(contract, key, args, name, businessObject = null) {
+
+    // Strip the key to just the method name (in cases where the method was called
+    // using the full method sig e.g. contract["draw(address,uint256)"](foo, bar)
+    const method = key.replace(/\(.*\)$/g, '');
+    const metadata = { contract: name, method, args };
+    let dsProxyAddress = null;
+
+    if (typeof args !== 'undefined' && Array.isArray(args) && typeof args[args.length - 1] === 'object') {
+
+      // Detect additional metadata attatched to last arg and merge it with default metadata
+      if (args[args.length - 1].hasOwnProperty('metadata')) {
+        Object.assign(metadata, args[args.length - 1].metadata);
+        delete args[args.length - 1].metadata;
+      }
+
+      // Detect proxy address and route through DSProxy contract
+      if (args[args.length - 1].hasOwnProperty('dsProxyAddress')) {
+        dsProxyAddress = args[args.length - 1].dsProxyAddress;
+        delete args[args.length - 1].dsProxyAddress;
+        this.get('log').debug('Using DSProxy ' + dsProxyAddress + ' for this tx');
+      }
+
+      // If last arg item is an empty object, remove it
+      if (Object.keys(args[args.length - 1]).length === 0) args.pop();
+    }
+
+    // DSProxy handling – different from the fact that this class is called Proxy ;)
+    if (dsProxyAddress !== null) {
+      const dsProxyContract = wrapContract(
+        new Contract(
+          dsProxyAddress,
+          dappHub.dsProxy,
+          this.get('web3').ethersSigner()
```

Comment:
instead of re-adding `ethersSigner` to `Web3Service`, please just use `this.get('web3').ethersProvider().getSigner()`